### ISSUE TITLE
`HTTPure.Contrib.Router` should probably be renamed

### ISSUE THREAD

**Issue by cprussin** [cprussin/purescript-httpure-rest-router#2]
tl;dr: I think we should probably try to make all contrib packages only export things in the `HTTPure.Contrib.MyPackage` scope that they're creating.

I'm thinking that `HTTPure.Contrib.Router` should be rolled into `HTTPure.Contrib.Rest.Router`, or possibly extracted to another module and named something like `PathBasedRouter` (this name is off the top of my head, we can probably do better).  I think that `HTTPure.Contrib.Router` is a super generic name that could be used for any number of modules that write a router, and that router may or may not be only based on paths (for instance, a module that routes requests only based on a query parameter could be used by a middleware that implements a developer console, and that module could easily be called `Router`).  So keeping it named `HTTPure.Contrib.Router` could end up causing collisions and sets a bad precedence for other contrib packages to create modules that are outside the scope they create...

Thougts?  Agree/disagree?
